account_balance_reporting/migrations/9.0.1.0.0/post-migration.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+def fill_dates(cr):
+    """Fill start/end dates, based on the periods
+    that were being used in 8.0"""
+
+    q_subquery = """
+        SELECT abr.id, min(ap.date_start) AS min_date_start,
+               max(ap.date_stop) AS max_date_stop
+        FROM account_balance_reporting_account_period_current_rel AS rel
+        JOIN account_balance_reporting AS abr
+        ON abr.id = rel.account_balance_reporting_id
+        JOIN account_period AS ap
+        ON ap.id = rel.period_id
+        WHERE abr.check_filter = 'periods'
+        GROUP BY abr.id
+    """
+    # Migrar periodos actuales
+    query_periodos_actuales = """
+        UPDATE account_balance_reporting as abr
+        SET current_date_from = Q.min_date_start,
+            current_date_to = Q.max_date_stop
+        FROM (%s) AS Q
+        where abr.id = Q.id
+    """ % q_subquery
+
+    cr.execute(query_periodos_actuales)
+
+    # Migrar ejercicio fiscal actual, si no hay periodos
+    query_ejercicio_actual = """
+        UPDATE account_balance_reporting as abr
+        SET current_date_from = fy.date_start,
+            current_date_to = fy.date_stop
+        FROM account_fiscalyear AS fy
+        WHERE abr.id not in (SELECT id from (%s) as Q)
```

Comment:
Haz mejor `LEFT JOIN account_balance_reporting_account_period_current_rel AS rel ON abr.id = rel.account_balance_reporting_id WHERE rel.account_balance_reporting_id IS NULL` y quitas esa pedazo subconsulta.

Reply:
No acabo de ver esta parte que dices. Hay que poner las fechas de account_fiscalyear solamente en el caso de que no se hayan informado ningun periodo.

Reply:
Claro, pero eso es lo que hace ese código. El `LEFT JOIN` hace que aparezcan los resultados aunque no pueda cruzar tablas por no haber datos, y la comprobación `IS NULL` precisamente deja solo los `abr` que no tienen periodos definidos.